vendor: upgrade twpayne/go-geom

otan opened the pull request:
This contains a critical GeometryCollection layout fix that allows
GEOMETRYCOLLECTION EMPTY to work.

Release justification: low-risk updates to new functionality

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/61096)
<!-- Reviewable:end -->

otan committed:
vendor: upgrade twpayne/go-geom

This contains a critical GeometryCollection layout fix that allows
GEOMETRYCOLLECTION EMPTY to work.

Release justification: low-risk updates to new functionality

Release note: None

otan commented:
bors r=andyyang890
